samples/testbot/test-bot/BotOrchestration.cs

```diff
@@ -0,0 +1,379 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Newtonsoft.Json;
+
+namespace DFTestBot
+{
+    public static class BotOrchestration
+    {
+        // NOTE: Using environment variables in orchestrator functions is not safe since environment variables are non-deterministic.
+        //       I'm ignoring this advice for now for the sake of expediency
+        static readonly Uri DeploymentServiceBaseUrl = new Uri(Environment.GetEnvironmentVariable("DEPLOYMENT_SERVICE_BASE_URL"));
+        static readonly string DeploymentServiceKey = Environment.GetEnvironmentVariable("DEPLOYMENT_SERVICE_API_KEY");
+
+        [FunctionName(nameof(BotOrchestrator))]
+        public static async Task BotOrchestrator(
+            [OrchestrationTrigger] IDurableOrchestrationContext context,
+            ILogger log)
+        {
+            log = context.CreateReplaySafeLogger(log);
+            
+            DateTime startTimeUtc = context.CurrentUtcDateTime;
+            TestParameters testParameters = context.GetInput<TestParameters>();
+
+            // Create a new resource group
+            if (!await TryCallDeploymentServiceHttpApiAsync("api/CreateNewResourceGroup", context, log, testParameters))
+            {
+                string message = $"Failed to create a new resource group! 💣 Check the internal deployment service logs for more details.";
+                await context.CallActivityAsync(nameof(PatchGitHubComment), (testParameters.GitHubCommentIdApiUrl, message));
+                throw new Exception(message);
+            }
```

Comment:
It seems to me that there's a pattern in this block that gets repeated throughout this orchestration. The pattern being:
```
if(!<await some task>):
    write comment
    throw exception
```

Maybe we could abstract this pattern into a helper function that gets called repeatedly with different params?
It could be called something like "CallDeploymentStep"? I think it would make the code more concise in places and make it easier to parse the sequence of steps taken by this orchestrator.

But I'd like to hear if others agree :)

Reply:
That could potentially clean things up a bit, but it may be smoother to have it take both an error message and an _optional_ success message to be posted to GitHub, as that is the more general case of the pattern from what I can tell.